src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -935,7 +935,7 @@ impl LinkCollector<'_, '_> {
             }
             .trim();
 
-            if path_str.contains(|ch: char| !(ch.is_alphanumeric() || ch == ':' || ch == '_')) {
+            if path_str.contains(|ch: char| !(ch.is_alphanumeric() || ch == ':' || ch == '_' || ch == '<' || ch == '>')) {
```

Comment:
nit: this is getting kind of long
```suggestion
            if path_str.contains(|ch: char| !(ch.is_alphanumeric() || ":_<>".contains(ch))) {
```